tests/test_init_output_run.py

```diff
@@ -0,0 +1,532 @@
+# to use; please see the files gpl-3.0.txt and/or bsd_license.txt,
+# respectively.  If you choose the GPL option then the following text applies
+# (but note that there is still no warranty even if you opt for BSD instead):
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import itertools
+import tempfile
+import unittest
+from collections.abc import Iterator
+from contextlib import contextmanager
+from typing import ClassVar
+
+import lsst.utils.tests
+from lsst.daf.butler import (
+    Butler,
+    DatasetRef,
+    DatasetType,
+    MissingDatasetTypeError,
+    QuantumBackedButler,
+    SerializedDatasetType,
+    StorageClassFactory,
+)
+from lsst.daf.butler.registry import ConflictingDefinitionError
+from lsst.pipe.base import QuantumGraph
+from lsst.pipe.base.all_dimensions_quantum_graph_builder import AllDimensionsQuantumGraphBuilder
+from lsst.pipe.base.pipeline_graph import PipelineGraph
+from lsst.pipe.base.tests.mocks import (
+    DynamicConnectionConfig,
+    DynamicTestPipelineTask,
+    DynamicTestPipelineTaskConfig,
+    MockDataset,
+)
+
+
+def _have_example_storage_classes() -> bool:
+    """Check whether some storage classes work as expected.
+
+    Given that these have registered converters, it shouldn't actually be
+    necessary to import those types in order to determine that they're
+    convertible, but the storage class machinery is implemented such that types
+    that can't be imported can't be converted, and while that's inconvenient
+    here it's totally fine in non-testing scenarios where you only care about a
+    storage class if you can actually use it.
+    """
+    getter = StorageClassFactory().getStorageClass
+    return (
+        getter("ArrowTable").can_convert(getter("ArrowAstropy"))
+        and getter("ArrowAstropy").can_convert(getter("ArrowTable"))
+        and getter("ArrowTable").can_convert(getter("DataFrame"))
+        and getter("DataFrame").can_convert(getter("ArrowTable"))
+    )
+
+
+class InitOutputRunTestCase(unittest.TestCase):
+    """Tests for the init_output_run methods of PipelineGraph and
+    QuantumGraph.
+    """
+
+    INPUT_COLLECTION: ClassVar[str] = "overall_inputs"
+
+    @contextmanager
+    def make_butler(self) -> Iterator[Butler]:
+        """Wrap a temporary local butler repository in a context manager."""
+        with tempfile.TemporaryDirectory() as root:
```

Comment:
Thanks for the reminder.  I actually intentionally went with `tempfile.TemporaryDirectory` over `lsst.utils.tests.temporaryDirectory` because I knew the standard library could now do that (and the only reason we invented our own was that the standard library couldn't before), and then I forgot to actually use that flag :man_facepalming:.